src/nanotron/models/llama.py

```diff
@@ -918,36 +947,129 @@ def init_model_randomly(self, config):
+                        return torch.nn.init.normal_(module.weight, mean=0.0, std=sigma / math.sqrt(2 * num_layers))
+
+                    if hasattr(module, "linear_type"):
+                        init_weight_using_mu_transfer(
+                            module.weight,
+                            name="weight",
+                            hidden_size=hidden_size,
+                            linear_type=module.linear_type,
+                            default=default,
+                        )
+                    else:
+                        default()
                 elif "bias" == param_name:
-                    param.zero_()
+
+                    def default():
+                        return param.zero_()
+
+                    if hasattr(module, "linear_type"):
+                        init_weight_using_mu_transfer(
+                            param,
+                            name="bias",
+                            hidden_size=hidden_size,
+                            linear_type=module.linear_type,
+                            default=default,
+                        )
+                    else:
+                        default()
                 else:
                     raise ValueError(f"Who the fuck is {param_name}?")
             elif isinstance(module, TritonRMSNorm):
                 if "weight" == param_name:
                     # TODO @thomasw21: Sometimes we actually want 0
-                    module.weight.fill_(1)
+                    def default():
+                        return module.weight.fill_(1)
+
+                    if hasattr(module, "linear_type"):
+                        init_weight_using_mu_transfer(
+                            module.weight,
+                            name="weight",
+                            hidden_size=hidden_size,
+                            linear_type=module.linear_type,
+                            default=default,
+                        )
+                    else:
+                        default()
                 elif "bias" == param_name:
-                    module.bias.zero_()
+
+                    def default():
+                        return module.bias.zero_()
+
+                    if hasattr(module, "linear_type"):
+                        init_weight_using_mu_transfer(
+                            module.bias,
+                            name="bias",
+                            hidden_size=hidden_size,
+                            linear_type=module.linear_type,
+                            default=default,
+                        )
+                    else:
+                        default()
                 else:
                     raise ValueError(f"Who the fuck is {param_name}?")
             elif isinstance(module, TensorParallelEmbedding):
-                nn.init.normal_(module.weight, mean=0.0, std=std)
+
+                def default():
+                    return nn.init.normal_(module.weight, mean=0.0, std=std)
```

Comment:
there's no need for default function here i think. You can just put it in "else"

Reply:
same for above